src/redux/session/selectors.js

```diff
@@ -39,6 +40,27 @@ export const BALANCES_COMPARATOR_URGENCY = (item1, item2) => {
   return s1 < s2 ? -1 : (s1 > s2 ? 1 : 0)
 }
 
+export const getProfileTokensList = () => createSelector(
+  [ getProfile, getTokens, getMainWallet ],
+  (profile, tokens, mainWallet) => {
+
+    const symbolToTitle = (symbol) => {
+      const titleList = [
+        { symbol: 'XEM', title: 'NEM' },
+      ]
+      const newTitle = titleList.find((s) => s.symbol === symbol)
+      return newTitle?newTitle.title:symbol
+    }
+
+    const addressesInWallet = mainWallet.addresses()
+    return rebuildProfileTokens(profile, tokens)
```

Comment:
Sometimes we may have no XEM or BTC token initialized but need a NEM or BTC address.
Please just hardcode ETH, BTC, BTG & NEM here.

Reply:
For example:
1) BTC or XEM middleware unavailable
2) BTC or XEM switched off from the user interface (hidden from profile, hidden from gallery).

You have in global scope:
EthereumDAO
NemDAO
BitcoinDAO (BTC)
BitcoinDAO (BTG)
it is all you need to get addresses & keys.